Source/Meadow.Foundation.Core/Sensors/Hid/AnalogJoystick.cs

```diff
@@ -371,5 +383,32 @@ float GetNormalizedPosition(Voltage value, bool isHorizontal)
 
             return (float)normalized;
         }
+
+        ///<inheritdoc/>
+        public void Dispose()
+        {
+            Dispose(disposing: true);
+            GC.SuppressFinalize(this);
+        }
+
+        /// <summary>
+        /// Dispose of the object
+        /// </summary>
+        /// <param name="disposing">Is disposing</param>
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!IsDisposed)
+            {
+                if (disposing && createdPorts)
+                {
+                    HorizontalInputPort?.StopUpdating();
+                    HorizontalInputPort?.Dispose();
```

Comment:
Does Dispose on the Port already call StopUpdating?  If not, it should, and you can eliminate these calls

Reply:
that's a good call - I shall check